test/e2e/issues_test.go

```diff
@@ -339,6 +339,54 @@ spec:
+		CreateNamespace(nsParent)
+		CreateNamespace(nsChild)
+		MustRun("kubectl hns set", nsChild, "--parent", nsParent)
+
+		// creare secret on parent namespace
+		MustRun("kubectl -n", nsParent, "create secret generic my-creds --from-literal=password=iamteamb")
+	
+		// after setting Propagate mode, the object shows up in the child namespace
+		MustRun("kubectl hns config set-resource secrets --mode Propagate --force")
+		RunShouldContain("my-creds", defTimeout, "kubectl -n", nsChild, "get secrets")
+
+		// after setting AllowPropagate mode, the object is removed from the child namespace
+		MustRun("kubectl hns config set-resource secrets --mode AllowPropagate --force")
+		RunShouldNotContain("my-creds", defTimeout, "kubectl -n", nsChild, "get secrets")
+
+		// after adding a Selector, the object shows up in the child namespace
+		MustRun("kubectl annotate secret my-creds -n", nsParent, "propagate.hnc.x-k8s.io/select="+nsChild+".tree.hnc.x-k8s.io/depth")
+		RunShouldContain("my-creds", defTimeout, "kubectl -n", nsChild, "get secrets")
+		// after removing a Selector, the object does not show up in the child namespace
+		MustRun("kubectl annotate secret my-creds -n", nsParent, "propagate.hnc.x-k8s.io/select-")
+		RunShouldNotContain("my-creds", defTimeout, "kubectl -n", nsChild, "get secrets")
+
+		// after adding a treeSelector, the object shows up in the child namespace
+		MustRun("kubectl annotate secret my-creds -n", nsParent, "propagate.hnc.x-k8s.io/treeSelect="+nsChild)
+		RunShouldContain("my-creds", defTimeout, "kubectl -n", nsChild, "get secrets")
+		// after removing a treeSelector, the object does not show up in the child namespace
+		MustRun("kubectl annotate secret my-creds -n", nsParent, "propagate.hnc.x-k8s.io/treeSelect-")
+		RunShouldNotContain("my-creds", defTimeout, "kubectl -n", nsChild, "get secrets")
+
+		// after adding a noneSelector, the object does not show up in the child namespace
+		MustRun("kubectl annotate secret my-creds -n", nsParent, "propagate.hnc.x-k8s.io/none=true")
+		RunShouldNotContain("my-creds", defTimeout, "kubectl -n", nsChild, "get secrets")
+		// after removing a noneSelector, the object does not show up in the child namespace
+		MustRun("kubectl annotate secret my-creds -n", nsParent, "propagate.hnc.x-k8s.io/none-")
+		RunShouldNotContain("my-creds", defTimeout, "kubectl -n", nsChild, "get secrets")
+
+		// after adding a allSelector, the object shows up in the child namespace
+		MustRun("kubectl annotate secret my-creds -n", nsParent, "propagate.hnc.x-k8s.io/all=true")
+		RunShouldContain("my-creds", defTimeout, "kubectl -n", nsChild, "get secrets")
+		// after removing a allSelector, the object does not show up in the child namespace
+		MustRun("kubectl annotate secret my-creds -n", nsParent, "propagate.hnc.x-k8s.io/all-")
+		RunShouldNotContain("my-creds", defTimeout, "kubectl -n", nsChild, "get secrets")
+
+		MustRun("kubectl hns config set-resource secrets --mode Ignore")
```

Comment:
Sorry I didn't notice this until now - if the test fails, this line will never run. Can you add it to the BeforeEach() function on line 23 of this file, and mention this test as the reason why this needs to be reset? That's my last comment, promise :)

Reply:
No problem, and all comments are welcome :)